### PR TITLE
Update transformers

### PR DESCRIPTION
**Pull request by maxjeblick**
This PR updates transformers version to 4.36.

By this, flash attention is supported (and enabled by default) natively, see https://twitter.com/efxmarty/status/1734931075367850385

### CODE REVIEW

**Review by psinger — CHANGES_REQUESTED** [review]
Overall, looks good, thx.

But let's discuss proper setting for config.

File: llm_studio/src/utils/modeling_utils.py [hunk 1]
```diff
@@ -677,17 +677,18 @@ def create_nlp_backbone(cfg, model_class=AutoModel) -> Any:
         try:
             import flash_attn  # noqa: F401
 
-            use_flash_attention_2 = cfg.training.use_flash_attention_2
+            # see https://github.com/fxmarty/transformers/
```
[the remark this answers is not in the document]
Comment (maxjeblick) [reply llm_studio/src/utils/modeling_utils.py:680]: I did it manually, happy to split it differently. Not sure if there's a commonly used convention for this.

File: llm_studio/src/utils/modeling_utils.py [hunk 2]
```diff
@@ -677,17 +677,18 @@ def create_nlp_backbone(cfg, model_class=AutoModel) -> Any:
...
+            # blob/3f06a3a0aec8cc1ec3ad6bf66ebe277392c5ab37/
+            # src/transformers/configuration_utils.py#L380
+            config._attn_implementation_internal = "flash_attention_2"
```
[the remark this answers is not in the document]
Comment (psinger) [reply llm_studio/src/utils/modeling_utils.py:683]: Got it, thx.
It seems one can pass it via the model constructor though:
https://huggingface.co/docs/transformers/perf_infer_gpu_one

**Review by psinger — APPROVED** [review]
@maxjeblick if you are confident this _internal setting is sufficient, let's merge
thx!!!

**Pull request merged**